doc/on_chain_spec.md

```diff
@@ -669,3 +669,175 @@ that distributing their respective portions of the auction fee
+but that can not be used to verify a signature produced by the bidders signing key.
+Unfortunately, it is not possible to even derive the `PubKeyHash` from the verification key on-chain,
+so we will have to carry around both in the datum.
+
+To do this, we must modify `BidTerms` with the bidder vkey,
+and the signature, representing the signed payload.
+
+```haskell
+data BidTerms = BidTerms
+  { bidderPKH :: PubKeyHash
+  -- ^ PubKeyHash of whoever submitted the bid
+  , bidderVK :: BuiltinByteString
+  -- ^ New field. Verification Key of whoever submitted the bid
+  , bidPrice :: Natural
+  -- ^ Which price did the bidder set to buy the auction lot?
+  , bidderSignature :: BuiltinByteString
+  -- ^ New field. Represents the signed payload by the bidder
+  , sellerSignature :: BuiltinByteString
+  -- ^ Represents the signed payload by the seller
+  }
+```
+
+This signature can be generated by the bidder,
+by signing with their signing key,
+a tuple consisting of the `hydraHeadId`, `bidderPKH` and `bidAmount`
+
+```haskell
+generateBidderSignature :: SigningKey PaymentKey -> HeadId -> Natural -> PubKeyHash -> BuiltinByteString
+generateBidderSignature skey headId bidAmount bidderPKH = toBuiltin $ dsign skey $ bidderSignatureMessage headId bidderPKH bidAmount
+
+bidderSignatureMessage :: HeadId -> Natural -> PubKeyHash -> BuiltinByteString
+bidderSignatureMessage headId bidAmount bidderPKH = toByteString headId <> toBysteString bidderPKH
+```
+
+We also need to extend the previous implementation of `sellerSignatureMessage` to also include the bidder verification key in the signature.
+We will use this to verify the association between bidder PKH and VK on-chain.
+
+```haskell
+generateSellerSignature :: SigningKey PaymentKey -> HeadId -> PubKeyHash -> SigningKey VerificationKey ->  BuiltinByteString
+generateSellerSignature skey = toBuiltin . (dsign skey) . sellerSignatureMessage
+
+sellerSignatureMessage :: HeadId -> PubKeyHash -> SigningKey VerificationKey -> BuiltinByteString
+sellerSignatureMessage headId bidderPKH bidderVK = toByteString headId <> toBysteString bidderPKH <> toByteString bidderVK
+```
+
+We now extend the Standing Bid Validator, to use on-chain verification
+to check that each new standing bid datum, is signed by the correct bidder.
+
+In particular, we can change `validNewBid` in the following way:
+
+```haskell
+validNewBid
+  :: AuctionTerms
+  -> StandingBidState
+  -> StandingBidState
+  -> Bool
+validNewBid AuctionTerms{..} oldStandingBidState newStandingBidState =
+  isJust $ do
+    let StandingBidState oldApprovedBidders oldBid = oldStandingBidState
+    let StandingBidState newApprovedBidders newBid = newStandingBidState
+    BidTerms bidderPKH bidderVK newPrice bidderSignature sellerSignature <- newBid
+    -- The seller has allowed the bidder to participate in the auction
+    guard $ verifyEd25519Signature (sellerVK terms) (sellerSignatureMessage (hydraHeadId terms) bidderPKH bidderVK) sellerSignature
```

Comment:
> use an on-chain primitive to verify something off-chain

I do not understand what you mean. I was talking about re-checking it off-chain, but on client side.

Reply:
Sorry, I was replying to this point specifically:

> This check should be separated for reuse

I agree with everything else you said.

What i meant is that `verifyEd25519Signature` is part of plutustx, the client side will already need to import something like `Crypto.Sign.Ed25519` to build the bidder signature, so it might as well use the `verify` function from there to to the check you mentioned